### ISSUE TITLE
System Error when running Ganache 2.1.2 on win32

### ISSUE THREAD

**Issue by chenAdmin** [trufflesuite/ganache#1586]
PLATFORM: win32
GANACHE VERSION: 2.1.2

EXCEPTION:
```
Error: EPERM: operation not permitted, rename 'C:\Users\admin\AppData\Roaming\Ganache\default\ContractCache.3943141732' -> 'C:\Users\admin\AppData\Roaming\Ganache\default\ContractCache'
    at Object.fs.renameSync (fs.js:772:18)
    at writeFileSync (C:\Program Files\WindowsApps\Ganache_2.1.2.0_x64__zh355ej5cj694\app\resources\app.asar\node_modules\write-file-atomic\index.js:124:8)
    at LocalStorage.setItem (C:\Program Files\WindowsApps\Ganache_2.1.2.0_x64__zh355ej5cj694\app\resources\app.asar\node_modules\node-localstorage\LocalStorage.js:188:7)
    at JsonStorage.setToStorage (C:\Program Files\WindowsApps\Ganache_2.1.2.0_x64__zh355ej5cj694\app\resources\app.asar\src\main\types\json\JsonStorage.js:72:20)
    at JsonStorage.set (C:\Program Files\WindowsApps\Ganache_2.1.2.0_x64__zh355ej5cj694\app\resources\app.asar\src\main\types\json\JsonStorage.js:78:12)
    at ContractCache.addEvent (C:\Program Files\WindowsApps\Ganache_2.1.2.0_x64__zh355ej5cj694\app\resources\app.asar\src\main\types\contracts\ContractCache.js:115:22)
    at TruffleIntegrationService.<anonymous> (C:\Program Files\WindowsApps\Ganache_2.1.2.0_x64__zh355ej5cj694\app\resources\app.asar\src\main\main.js:222:33)
    at emitOne (events.js:115:13)
    at TruffleIntegrationService.emit (events.js:210:7)
    at ChildProcess.child.on.message (C:\Program Files\WindowsApps\Ganache_2.1.2.0_x64__zh355ej5cj694\app\resources\app.asar\src\common\services\TruffleIntegrationService.js:66:14)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at emit (internal/child_process.js:768:12)
    at _combinedTickCallback (internal/process/next_tick.js:141:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

APPLICATION LOG:
```
T+4465450ms: eth_blockNumber
T+4485483ms: eth_blockNumber
T+4505505ms: eth_blockNumber
T+4525541ms: eth_blockNumber
T+4545575ms: eth_blockNumber
T+4565604ms: eth_blockNumber
T+4585627ms: eth_blockNumber
T+4605642ms: eth_blockNumber
T+4625680ms: eth_blockNumber
T+4645716ms: eth_blockNumber
T+4665764ms: eth_blockNumber
T+4685800ms: eth_blockNumber
T+4705832ms: eth_blockNumber
T+4725864ms: eth_blockNumber
T+4745891ms: eth_blockNumber
T+4765925ms: eth_blockNumber
T+4785949ms: eth_blockNumber
T+4805969ms: eth_blockNumber
T+4826006ms: eth_blockNumber
T+4846027ms: eth_blockNumber
T+4866052ms: eth_blockNumber
T+4886076ms: eth_blockNumber
T+4906097ms: eth_blockNumber
T+4926131ms: eth_blockNumber
T+4946156ms: eth_blockNumber
T+4966191ms: eth_blockNumber
T+4986224ms: eth_blockNumber
T+5006259ms: eth_blockNumber
T+5026294ms: eth_blockNumber
T+5046334ms: eth_blockNumber
T+5066368ms: eth_blockNumber
T+5086395ms: eth_blockNumber
T+5106414ms: eth_blockNumber
T+5126446ms: eth_blockNumber
T+5146479ms: eth_blockNumber
T+5166517ms: eth_blockNumber
T+5186552ms: eth_blockNumber
T+5206573ms: eth_blockNumber
T+5226610ms: eth_blockNumber
T+5246645ms: eth_blockNumber
T+5266671ms: eth_blockNumber
T+5286691ms: eth_blockNumber
T+5306710ms: eth_blockNumber
T+5326732ms: eth_blockNumber
T+5346757ms: eth_blockNumber
T+5366792ms: eth_blockNumber
T+5386819ms: eth_blockNumber
T+5406841ms: eth_blockNumber
T+5426862ms: eth_blockNumber
T+5446886ms: eth_blockNumber
T+5466910ms: eth_blockNumber
T+5486936ms: eth_blockNumber
T+5506957ms: eth_blockNumber
T+5526984ms: eth_blockNumber
T+5547004ms: eth_blockNumber
T+5567037ms: eth_blockNumber
T+5587068ms: eth_blockNumber
T+5607091ms: eth_blockNumber
T+5627115ms: eth_blockNumber
T+5647149ms: eth_blockNumber
T+5667175ms: eth_blockNumber
T+5687212ms: eth_blockNumber
T+5707239ms: eth_blockNumber
T+5727274ms: eth_blockNumber
T+5747298ms: eth_blockNumber
T+5767320ms: eth_blockNumber
T+5787345ms: eth_blockNumber
T+5807380ms: eth_blockNumber
T+5827416ms: eth_blockNumber
T+5847439ms: eth_blockNumber
T+5867473ms: eth_blockNumber
T+5887496ms: eth_blockNumber
T+5907520ms: eth_blockNumber
T+5927554ms: eth_blockNumber
T+5947576ms: eth_blockNumber
T+5967599ms: eth_blockNumber
T+5987620ms: eth_blockNumber
T+6007645ms: eth_blockNumber
T+6027676ms: eth_blockNumber
T+6047700ms: eth_blockNumber
T+6067723ms: eth_blockNumber
T+6087758ms: eth_blockNumber
T+6107794ms: eth_blockNumber
T+6127831ms: eth_blockNumber
T+6147866ms: eth_blockNumber
T+6167894ms: eth_blockNumber
T+6187918ms: eth_blockNumber
T+6207953ms: eth_blockNumber
T+6227985ms: eth_blockNumber
T+6248018ms: eth_blockNumber
T+6268052ms: eth_blockNumber
T+6288087ms: eth_blockNumber
T+6308124ms: eth_blockNumber
T+6328157ms: eth_blockNumber
T+6348181ms: eth_blockNumber
T+6368204ms: eth_blockNumber
T+6388224ms: eth_blockNumber
T+6408260ms: eth_blockNumber
T+6428294ms: eth_blockNumber
T+6448318ms: eth_blockNumber
T+6468331ms: eth_blockNumber
T+6488353ms: eth_blockNumber
T+6508386ms: eth_blockNumber
T+6528409ms: eth_blockNumber
T+6548443ms: eth_blockNumber
T+6568478ms: eth_blockNumber
T+6588497ms: eth_blockNumber
T+6608531ms: eth_blockNumber
T+6628548ms: eth_blockNumber
T+6648561ms: eth_blockNumber
T+6668585ms: eth_blockNumber
T+6688610ms: eth_blockNumber
T+6708645ms: eth_blockNumber
T+6728670ms: eth_blockNumber
T+6748695ms: eth_blockNumber
T+6768720ms: eth_blockNumber
T+6788744ms: eth_blockNumber
T+6808770ms: eth_blockNumber
T+6828785ms: eth_blockNumber
T+6848820ms: eth_blockNumber
T+6868831ms: eth_blockNumber
T+6888867ms: eth_blockNumber
T+6908894ms: eth_blockNumber
T+6928913ms: eth_blockNumber
T+6948946ms: eth_blockNumber
T+6968980ms: eth_blockNumber
T+6989016ms: eth_blockNumber
T+7009053ms: eth_blockNumber
T+7029088ms: eth_blockNumber
T+7049099ms: eth_blockNumber
T+7069123ms: eth_blockNumber
T+7089148ms: eth_blockNumber
```

**Comment by davidmurdoch** [comment]
Should be fixed in https://github.com/trufflesuite/ganache/releases/tag/v2.3.0! Please let me know if you are still experiencing problems!

**Comment by wbt** [comment]
For those tracing through...this [was not](https://github.com/trufflesuite/ganache/issues/1721) fixed in 2.3.0.
